### PR TITLE
feat(leet): add custom axis support for remote runs from define_metric

### PR DESCRIPTION
**Pull request by jacobromero**
Description
-----------

What does the PR do? Include a concise description of the PR contents.

This PR adds custom axis support from `define_metric` for remote runs. which have already been uploaded and pulled from the W&B backend. It aligns support for local runs added in a previous [PR](https://github.com/wandb/wandb/pull/12568).

- [x] I updated CHANGELOG.unreleased.md, or it's not applicable

Testing
-------
How was this PR tested?

#### Test script

```python
import math

import wandb

STEPS_PER_EPOCH = 5

with wandb.init(project="leettest") as run:
    # Use "epoch" as the x-axis instead of the default _step.
    run.define_metric("epoch", hidden=True)
    run.define_metric("loss", step_metric="epoch")

    for step in range(500):
        # _step advances every log(); epoch advances once per STEPS_PER_EPOCH logs.
        epoch = step / STEPS_PER_EPOCH
        loss = math.exp(-epoch / 20) + 0.1 * math.sin(epoch)
        run.log({"epoch": epoch, "loss": loss})
```

#### Wandb leet with remote run

image

### CODE REVIEW

**Comment by dmitryduev** [comment]
side note: `run.define_metric("epoch", hidden=True)` -- should we actually comply with that? :)

**Comment by jacobromero** [comment]
> side note: `run.define_metric("epoch", hidden=True)` -- should we actually comply with that? :)

That makes sense to me :slightly_smiling_face:  
Local runs weren't followng this either so this was to align the behavior. But happen to include the fix for this behavior here as well
